### ISSUE TITLE
Can't build for WebAssembly

### ISSUE THREAD

**Issue by jscheid** [sdd/kiddo_v1#12]
I'd like to use this crate on the web but getting errors. Are there any plans to support platforms other than x86_64?

```
$ wasm-pack build --release --target web rust
[INFO]: 🎯  Checking for the Wasm target...
[INFO]: 🌀  Compiling to Wasm...
   Compiling kiddo v0.1.6
error[E0432]: unresolved import `std::arch::x86_64`
 --> /redacted/.cargo/registry/src/github.com-1ecc6299db9ec823/kiddo-0.1.6/src/distance.rs:6:16
  |
6 | use std::arch::x86_64::*;
  |                ^^^^^^ could not find `x86_64` in `arch`

error[E0412]: cannot find type `__m128` in this scope
  --> /redacted/.cargo/registry/src/github.com-1ecc6299db9ec823/kiddo-0.1.6/src/distance.rs:10:11
   |
10 |     simd: __m128,
   |           ^^^^^^ not found in this scope

error[E0425]: cannot find function `dot_sse` in this scope
  --> /redacted/.cargo/registry/src/github.com-1ecc6299db9ec823/kiddo-0.1.6/src/distance.rs:78:14
   |
78 |     unsafe { dot_sse(ap, bp) }
   |              ^^^^^^^ not found in this scope

error[E0425]: cannot find function `dot_sse` in this scope
  --> /redacted/.cargo/registry/src/github.com-1ecc6299db9ec823/kiddo-0.1.6/src/distance.rs:82:14
   |
82 |     unsafe { dot_sse(a.as_ptr(), b.as_ptr()) }
   |              ^^^^^^^ not found in this scope

error[E0425]: cannot find function `dot_sse_aligned` in this scope
  --> /redacted/.cargo/registry/src/github.com-1ecc6299db9ec823/kiddo-0.1.6/src/distance.rs:88:14
   |
88 |     unsafe { dot_sse_aligned(ap, bp) }
   |              ^^^^^^^^^^^^^^^ not found in this scope

error[E0740]: unions may not contain fields that need dropping
  --> /redacted/.cargo/registry/src/github.com-1ecc6299db9ec823/kiddo-0.1.6/src/distance.rs:10:5
   |
10 |     simd: __m128,
   |     ^^^^^^^^^^^^
   |
note: `std::mem::ManuallyDrop` can be used to wrap the type
  --> /redacted/.cargo/registry/src/github.com-1ecc6299db9ec823/kiddo-0.1.6/src/distance.rs:10:5
   |
10 |     simd: __m128,
   |     ^^^^^^^^^^^^

Some errors have detailed explanations: E0412, E0425, E0432, E0740.
For more information about an error, try `rustc --explain E0412`.
error: could not compile `kiddo` due to 6 previous errors
warning: build failed, waiting for other jobs to finish...
error: build failed
Error: Compiling your crate to WebAssembly failed
Caused by: failed to execute `cargo build`: exited with exit status: 101
  full command: "cargo" "build" "--lib" "--release" "--target" "wasm32-unknown-unknown"
$ rustc --version
rustc 1.58.0-nightly (e90c5fbbc 2021-11-12)
$
```

**Comment by sdd** [comment]
Thanks for the contribution Julian! I had to amend your commit to prefix it with `fix:` so that the semantic release pipeline would trigger a new release version and publish it to crates.io. It's just been published as v0.1.7.

I'm interested to find out how you get on using it on the web via wasm. 

Thanks again - Scott

**Comment by jscheid** [comment]
Thanks! It works well with wasm.
